homeassistant/components/sensor/seventeentrack.py

```diff
@@ -0,0 +1,274 @@
+        sensors.append(SeventeenTrackPackageSensor(data, package))
+
+    async_add_entities(sensors, True)
+
+
+class SeventeenTrackSummarySensor(Entity):
+    """Define a summary sensor."""
+
+    def __init__(self, data, status, initial_state):
+        """Initialize."""
+        self._attrs = {ATTR_ATTRIBUTION: DEFAULT_ATTRIBUTION}
+        self._data = data
+        self._state = initial_state
+        self._status = status
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        return self._attrs
+
+    @property
+    def icon(self):
+        """Return the icon."""
+        return 'mdi:package'
+
+    @property
+    def name(self):
+        """Return the name."""
+        return 'Packages {0}'.format(self._status)
+
+    @property
+    def state(self):
+        """Return the state."""
+        return self._state
+
+    @property
+    def unique_id(self):
+        """Return a unique, HASS-friendly identifier for this entity."""
+        return 'summary_{0}_{1}'.format(
+            self._data.account_id, slugify(self._status))
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit the value is expressed in."""
+        return 'packages'
+
+    async def async_update(self):
+        """Update the sensor."""
+        await self._data.async_update()
+
+        self._state = self._data.summary[self._status]
+
+
+class SeventeenTrackPackageSensor(Entity):
+    """Define an individual package sensor."""
+
+    def __init__(self, data, package):
+        """Initialize."""
+        self._attrs = {
+            ATTR_ATTRIBUTION: DEFAULT_ATTRIBUTION,
+            ATTR_DESTINATION_COUNTRY: package.destination_country,
+            ATTR_INFO_TEXT: package.info_text,
+            ATTR_LOCATION: package.location,
+            ATTR_ORIGIN_COUNTRY: package.origin_country,
+            ATTR_PACKAGE_TYPE: package.package_type,
+            ATTR_TRACKING_INFO_LANGUAGE: package.tracking_info_language,
+        }
+        self._data = data
+        self._state = package.status
+        self._tracking_number = package.tracking_number
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        return self._attrs
+
+    @property
+    def icon(self):
+        """Return the icon."""
+        return 'mdi:package'
+
+    @property
+    def name(self):
+        """Return the name."""
+        return self._tracking_number
+
+    @property
+    def state(self):
+        """Return the state."""
+        return self._state
+
+    @property
+    def unique_id(self):
+        """Return a unique, HASS-friendly identifier for this entity."""
+        return 'package_{0}_{1}'.format(
+            self._data.account_id, self._tracking_number)
+
+    async def async_update(self):
+        """Update the sensor."""
+        await self._data.async_update()
+
+        try:
+            package = next(
+                iter([
```

Comment:
Make this a generator directly by using parenthesis instead of brackets.
```py
(p for p in self._data.packages if ...)
```
Then we don't need `iter`.